src/AssocArray.php

```diff
@@ -216,6 +217,32 @@ public function withoutElement($value): AssocArray
         return $this->filter(Filters::notEqual($value));
     }
 
+    /**
+     * @phpstan-return IterableValue<TKey, TValue>
+     */
+    public function toIterableValue(): IterableValue
+    {
+        return new InfiniteIterableValue($this->toAssocArray());
+    }
+
+    /**
+     * @phpstan-return IterableValue<TValue, TKey>
+     */
+    public function flip(): IterableValue
+    {
+        return $this->toIterableValue()->flip();
```

Comment:
Why not through `GW\Value\Associable\Flip`? I don't think that `AssocArray` should return `IterableValue`.

Reply:
It is hard to flip AssocValue, because TKey must be int|string and TValue is mixed - therefore it produces phpstan type errors. I think it can be fully fixed only after rewrite AssocValue to some kind of map with mixed key and mixed value. 

For now I've changed it to:

```php
    /**
     * @phpstan-return AssocValue<int|string, TKey>
     */
    public function flip(): AssocValue;
```

it will be `AssocValue` - `AssocValue<TValue,TKey>` but `AssocValue<int|string, TKey>`

in other words - you loose type for key - from `Wrap::assoc([0 => 'xxx'])` as `AssocValue<int,string>` flip makes `AssocValue<int|string,int>`, not `AssocValue<string,int>`